python/pyspark/errors/exceptions/base.py

```diff
@@ -31,6 +34,7 @@ def __init__(
         message: Optional[str] = None,
         error_class: Optional[str] = None,
         message_parameters: Optional[Dict[str, str]] = None,
+        data: Optional[Iterable["Row"]] = None,
```

Comment:
Will this be useful for any other exceptions other than `PySparkAssertionError`?

Reply:
I initially added this parameter as a member of the super class(`PySparkException`) considering future expansion. For example, we could consider the case of later extending this member in `PySparkValueError` or `PySparkTypeError` to find problematic values ​​or types.

But it's true that this is currently only used for the logic related to `PySparkAssertionError`.

Can I ask you opinion, @HyukjinKwon ? Should we keep this parameter here as it is, or move into `PySparkAssertionError`?

Reply:
Actually good point. we can add it to `PySparkAssertionError` for now, and extend it later if we want

Reply:
Updated. Thanks!